crates/api/src/peer_meta_store.rs

```diff
@@ -29,6 +29,12 @@ pub trait PeerMetaStore: 'static + Send + Sync + std::fmt::Debug {
         key: String,
     ) -> BoxFuture<'_, K2Result<Option<bytes::Bytes>>>;
 
+    /// Get all peers for a given key.
+    fn get_all_by_key(
```

Comment:
A new function for the peer meta store to get all URLs with values for a given meta key, e. g. to get all unresponsive URLs.